python/src/spark_rapids_ml/core.py

```diff
@@ -484,6 +513,52 @@ def _fit_array_order(self) -> _ArrayOrder:
         return "F"
 
 
+class _FitMultipleIterator(Generic[CumlModel]):
+    """
+    Used by default implementation of Estimator.fitMultiple to produce models in a thread safe
+    iterator. This class handles the gpu versioned of fitMultiple where all param maps should be
+    fit in a single pass.
+
+    Parameters
+    ----------
+    fitMultipleModels : function
+        Callable[[], CumlModel] which fits multiple models to a dataset in a single pass.
+    numModels : int
+        Number of models this iterator should produce.
+
+    Notes
+    -----
+    See :py:meth:`Estimator.fitMultiple` for more info.
+    """
+
+    def __init__(
+        self, fitMultipleModels: Callable[[], List[CumlModel]], numModels: int
+    ):
+        """ """
```

Comment:
Is this empty docstring intentional?

Reply:
Oops, I didn't realize it. I copied it from pyspark.  Let me remove the empty docstring.

Reply:
Sry, new commit has removed it.